### PR TITLE
feat: Add bean deserialization support for executeJs return values

### PR DESCRIPTION
**Pull request by Artur-**
- Extend JacksonCodec.decodeAs() to deserialize JSON objects into Java beans using Jackson
- Extend JsonCodec.decodeAs() with similar bean deserialization support for consistency
- Add comprehensive unit tests for simple beans, nested beans, null handling, and error cases
- Add default constructors to test bean classes for Jackson compatibility
- Add integration tests with new buttons for testing bean return values from client
- Add ExecJavaScriptIT tests to verify complete client-to-server bean deserialization flow
- Maintain full backward compatibility for existing primitive type decoding
- Support both simple and complex nested bean structures returned from JavaScript

This enables JavaScript code to return complex objects that are automatically
deserialized into typed Java beans on the server side.

### CODE REVIEW

File: flow-server/src/test/java/com/vaadin/flow/internal/JacksonCodecTest.java [hunk 1]
```diff
@@ -371,6 +369,9 @@ private static class SimpleBean {
         public String text;
         public int value;
 
+        public SimpleBean() {
+        }
+
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/internal/JacksonCodecTest.java:374]: More of a question on what would we suggest people use when making beans. Having a no args constructor or adding the annotations like:
```
    private static class SimpleBean {
        public String text;
        public int value;

        @JsonCreator
        public SimpleBean(@JsonProperty("text") String text,
                          @JsonProperty("value") int value) {
            this.text = text;
            this.value = value;
        }
    }
```

Comment (Legioth) [reply flow-server/src/test/java/com/vaadin/flow/internal/JacksonCodecTest.java:374]: I'd say either beans with setters+getters or then records that just work out of the box.

Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/internal/JacksonCodecTest.java:374]: records is the way to go in most cases

**Review by caalador — APPROVED** [review]

File: flow-server/src/main/java/com/vaadin/flow/internal/JacksonCodec.java [hunk 2]
```diff
@@ -266,8 +267,15 @@ public static <T> T decodeAs(JsonNode json, Class<T> type) {
         } else if (JsonNode.class.isAssignableFrom(type)) {
             return type.cast(json);
         } else {
-            throw new IllegalArgumentException(
-                    "Unknown type " + type.getName());
+            // Try to deserialize as a bean using Jackson
```
Comment (Legioth) [inline flow-server/src/main/java/com/vaadin/flow/internal/JacksonCodec.java:270]: Any reason to still have custom code for the other cases above rather than just doing it all through Jackson?

File: flow-server/src/test/java/com/vaadin/flow/internal/JacksonCodecTest.java [hunk 3]
```diff
@@ -391,10 +395,74 @@ private static class OuterBean {
...
+            JacksonCodec.decodeAs(invalidJson, SimpleBean.class);
+            Assert.fail("Should have thrown IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(
+                    e.getMessage().contains("Cannot deserialize JSON to type"));
+        }
+    }
+
+    @Test
+    public void testDecodeAsPreservesExistingBehavior() {
```
Comment (Legioth) [inline flow-server/src/test/java/com/vaadin/flow/internal/JacksonCodecTest.java]: Test name makes no sense outside the context of this PR

Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/internal/JacksonCodecTest.java]: Renamed

File: flow-server/src/main/java/com/vaadin/flow/internal/JacksonCodec.java (hunk 2 above)
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/internal/JacksonCodec.java:270]: Yes and no. Handled in #22393
